File: solutions/fsharp/darts/1/Darts.fs
module Darts

let score (x: double) (y: double) : int =
    match System.Double.Hypot(x, y) with
    | d when d <= 1 -> 10
    | d when d <= 5 -> 5
    | d when d <= 10 -> 1
    | _ -> 0

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {
        double distance = Math.hypot(xOfDart, yOfDart);

        if (distance > 10) return 0;
        if (distance > 5)  return 1;
        if (distance > 1)  return 5;
        return 10;
    }
}
